### PR TITLE
Move MLflow Pipelines dependencies to core

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Move MLflow Pipelines dependencies to core.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [x] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -149,15 +149,6 @@ def run(self):
             # as part of the MLflow server monitoring add-on
             "prometheus-flask-exporter",
         ],
-        "pipelines": [
-            "scikit-learn>=1.0",
```
Comment (dbczumar) [inline setup.py:153]: We may have to make some small changes to support `scikit-learn<1.0` or restrict MLflow 2.0 to `scikit-learn>=1.0`

**Comment by harupy** [comment]
Looks like a test for dependency inference failed because mlflow now depends on sklearn:

```
_____________________________ test_prune_packages ______________________________

    def test_prune_packages():
        assert _prune_packages(["mlflow"]) == {"mlflow"}
        assert _prune_packages(["mlflow", "packaging"]) == {"mlflow"}
>       assert _prune_packages(["mlflow", "scikit-learn"]) == {"mlflow", "scikit-learn"}
E       AssertionError: assert {'mlflow'} == {'mlflow', 'scikit-learn'}
E         Extra items in the right set:
E         'scikit-learn'
E         Full diff:
E         - {'scikit-learn', 'mlflow'}
E         + {'mlflow'}
```

This can be fixed easily.

File: requirements/core-requirements.yaml [hunk 2]
```diff
@@ -46,3 +46,31 @@ waitress:
...
+  pip_release: ipython
+  max_major_version: 8
+
+markdown:
+  pip_release: markdown
+  max_major_version: 3
+
+Jinja2:
+  pip_release: Jinja2
+  max_major_version: 3
```
Comment (dbczumar) [inline requirements/core-requirements.yaml:76]: Looks like we aren't defining minimum versions for these anymore. Shall we?

**Review by dbczumar — COMMENTED** [review]
Thanks @harupy ! A couple small comments. Can we update the docs as well to indicate that MLP won't need to be installed separately?

File: requirements/core-requirements.yaml (hunk 2 above)
Comment (harupy) [reply requirements/core-requirements.yaml]: I forgot to set `minimum`. Fixed.

File: setup.py (hunk 1 above)
Comment (harupy) [reply setup.py:153]: Why are we restricting scikit-learn `>=1.0` in MLP?

Comment (harupy) [reply setup.py:153]: Is this because of `get_feature_names_out`?

https://github.com/mlflow/mlflow/blob/f65429478255b88a211d7dd262c159b4939b44b2/mlflow/pipelines/steps/transform.py#L28

Comment (harupy) [reply setup.py:153]: We could run MLP tests with `scikit-learn < 1.0`.

File: docs/source/pipelines.rst [hunk 3]
```diff
@@ -24,30 +24,6 @@ ML engineers, and DevOps teams to iterate very quickly within their domains of e
...
-
-.. note::
-  You can install MLflow Pipelines from a Databricks Notebook by running
-  ``%pip install mlflow[pipelines]`` or install MLflow Pipelines on a Databricks Cluster
-  by following the instructions at
-  https://docs.databricks.com/libraries/cluster-libraries.html#install-a-library-on-a-cluster.
-
-  Note that `Databricks Runtime <https://docs.databricks.com/runtime/dbr.html>`_ version 11.0
-  or greater is required in order to install MLflow Pipelines on Databricks.
-
```
Comment (harupy) [inline docs/source/pipelines.rst:50]: @dbczumar Removed this section.

Comment (harupy) [reply docs/source/pipelines.rst:50]:
```
.. note::
  MLflow Pipelines requires Make, which may not be preinstalled on some systems (e.g. Windows).
  Please ensure Make is installed before using MLflow Pipelines.
```

File: docs/source/pipelines.rst [hunk 4]
```diff
@@ -24,30 +24,11 @@ ML engineers, and DevOps teams to iterate very quickly within their domains of e
...
-  Please ensure Make is installed before using MLflow Pipelines.
-
-.. note::
-  You can install MLflow Pipelines from a Databricks Notebook by running
-  ``%pip install mlflow[pipelines]`` or install MLflow Pipelines on a Databricks Cluster
-  by following the instructions at
-  https://docs.databricks.com/libraries/cluster-libraries.html#install-a-library-on-a-cluster.
-
-  Note that `Databricks Runtime <https://docs.databricks.com/runtime/dbr.html>`_ version 11.0
-  or greater is required in order to install MLflow Pipelines on Databricks.
```
Comment (harupy) [inline docs/source/pipelines.rst:49]: Does this section need to be kept for Databrick Runtimes where mlflow>=2.0 is not pre-installed?

Comment (harupy) [reply docs/source/pipelines.rst:49]:
```
.. note::
  You can install MLflow Pipelines from a Databricks Notebook by running
  ``%pip install -U mlflow`` or install MLflow Pipelines on a Databricks Cluster
  by following the instructions at
  https://docs.databricks.com/libraries/cluster-libraries.html#install-a-library-on-a-cluster.

  Note that `Databricks Runtime <https://docs.databricks.com/runtime/dbr.html>`_ version 11.0
  or greater is required in order to install MLflow Pipelines on Databricks.
```

File: setup.py (hunk 1 above)
Comment (dbczumar) [reply setup.py:153]: Yes, it's because of `get_feature_names_out`.

File: docs/source/pipelines.rst (hunk 4 above)
Comment (dbczumar) [reply docs/source/pipelines.rst]: I think we should rephrase this to say that MLP requires DBR 11+, MLP is preinstalled on MLR 12+, and users can install MLP in DBR 11 via `pip install mlflow>=2.0` or `pip install mlflow[pipelines]<2.0` (if they want to remain on a 1.x version).

File: setup.py (hunk 1 above)
Comment (harupy) [reply setup.py:153]: If so, can we just remove `minimum: "1.0"`?

File: docs/source/pipelines.rst (hunk 4 above)
Comment (harupy) [reply docs/source/pipelines.rst]: Sounds good!

File: setup.py (hunk 1 above)
Comment (dbczumar) [reply setup.py:153]: Sounds good!

Comment (dbczumar) [reply setup.py:153]: @harupy Did we test this manually?

**Review by dbczumar — APPROVED** [review]
LGTM once we manually verify support for sklearn < 1. Thanks @harupy !

File: setup.py (hunk 1 above)
Comment (harupy) [reply setup.py:153]: @dbczumar I did in https://github.com/mlflow/mlflow/pull/6773.
